gammapy/irf/core.py

```diff
@@ -201,8 +202,20 @@ def evaluate(self, method=None, **kwargs):
             coord = kwargs.get(key, value)
             if coord is not None:
                 coords_default[key] = u.Quantity(coord, copy=False)
+        data = self._interpolate(coords_default.values(), method=method)
 
-        return self._interpolate(coords_default.values(), method=method)
+        if self.fill_value != self.default_interp_kwargs["fill_value"]:
+            idxs = self.axes.coord_to_idx(coords_default, clip=False)
+            invalid = np.broadcast_arrays(*[idx == -1 for idx in idxs])
+            mask = self._mask_out_bounds(invalid)
+            if not data.shape:
+                mask = mask.squeeze()
+            data[mask] = self.fill_value
+            data[~np.isfinite(data)] = self.fill_value
```

Comment:
Hm, within the defined data rang the IRFs should be valid I guess, why this additional masking?

Reply:
interp_missing_data can also add NaN, I thought that it would be better to keep the NaN there instead of fill_value as it modifies directly the data and then fix it on evaluate.